tests/threshold_sign.rs

```diff
@@ -86,24 +82,29 @@ where
 
     for size in sizes {
         let num_faulty_nodes = util::max_faulty(size);
-        let num_good_nodes = size - num_faulty_nodes;
         info!(
             "Network size: {} good nodes, {} faulty nodes",
-            num_good_nodes, num_faulty_nodes
+            size - num_faulty_nodes,
+            num_faulty_nodes
         );
         let unique_id: u64 = rng.gen();
         let mut count_true = 0;
         let mut count_false = 0;
         for i in 0..num_samples {
-            let adversary = |_| new_adversary(num_good_nodes, num_faulty_nodes);
             let nonce = format!("My very unique nonce {:x}:{}", unique_id, i);
             info!("Nonce: {}", nonce);
-            let new_coin = |netinfo: _| {
-                ThresholdSign::new_with_document(netinfo, nonce.clone())
-                    .expect("Failed to set the new coin's ID")
-            };
-            let network = TestNetwork::new(num_good_nodes, num_faulty_nodes, adversary, new_coin);
-            let coin = test_threshold_sign(network).parity();
+            let (mut net, _) = NetBuilder::new(0..size as u16)
+                .num_faulty(num_faulty_nodes as usize)
+                .message_limit(size * (size - 1))
+                .no_time_limit()
+                .adversary(new_adversary())
+                .using(move |node_info: NewNodeInfo<_>| {
+                    ThresholdSign::new_with_document(Arc::new(node_info.netinfo), nonce.clone())
+                        .expect("Failed to create a ThresholdSign instance.")
+                })
+                .build(&mut rng)
+                .expect("Could not construct test network.");
+            let coin = net.test_threshold_sign().parity();
```

Comment:
Not your fault, but now that we've replaced the `Coin` (uses threshold signatures as random values) with `ThresholdSign` itself, maybe the test should also verify that the signature is valid, probably inside `test_threshold_sign`. (Feel free to just add a `TODO` for now.)

Reply:
Added a TODO